nsv13/code/modules/overmap/ai-skynet.dm

```diff
@@ -1285,14 +1292,24 @@ Seek a ship thich we'll station ourselves around
 		if(will_use_ammo)
 			ammo_use++
 		did_fire = TRUE
-		next_firetime_gunspecific["[iter]"] = world.time + SW.fire_delay
-		if(SW.fire_delay < smallest_cooldown)
-			smallest_cooldown = SW.fire_delay
+		SW.next_firetime = world.time + SW.fire_delay + SW.ai_fire_delay
 
 	if(did_fire)
 		shots_left -= ammo_use
-		next_firetime = world.time + 1 SECONDS + smallest_cooldown
 		handle_cloak(CLOAK_TEMPORARY_LOSS)
+
+// Not as good as a carrier, but something
+/obj/structure/overmap/proc/ai_self_resupply()
+	ai_resupply_scheduled = FALSE
+	var/resupply_amount = CEILING(initial(missiles) / 4, 1)
+	if(missiles < resupply_amount)
+		missiles += resupply_amount
+	resupply_amount = CEILING(initial(torpedoes) / 4, 1)
+	if(torpedoes < resupply_amount)
+		torpedoes += resupply_amount
+	resupply_amount = CEILING(initial(shots_left) / 2, 1)
+	if(shots_left < resupply_amount)
+		shots_left += resupply_amount
```

Comment:
Why not just make it `missiles = CLAMP(missiles + initial(missiles)/4, 1, initial(missiles)/4)` etc. for other weapons instead of this?